includes/class-wc-payments-account.php

```diff
@@ -2767,6 +2781,24 @@ public function create_embedded_account_session(): array {
 		];
 	}
 
+	/**
+	 * Marks the account deletion in progress.
+	 *
+	 * @return void
+	 */
+	public function set_account_deletion_in_progress(): void {
+		$this->database_cache->add( Database_Cache::ACCOUNT_KEY, [] );
+	}
+
+	/**
+	 * Clears the account deletion in progress transient.
+	 *
+	 * @return void
+	 */
+	public function clear_account_deletion_in_progress(): void {
+		$this->get_cached_account_data( true );
+	}
+
```

Comment:
With the latest changes suggested by @marcinbot, these are not needed anymore, @oaratovskyi . We don't track the progress, but force the account cache into a deleted state before doing the actual deletion.